### ISSUE TITLE
Mention the need to pull tags when performing a development install

### ISSUE THREAD

**Issue by brandonwillard** [aesara-devs/aesara#1457]
We need to mention that users should pull the tags (e.g. `git fetch -t`) when attempting a developer build; otherwise, they could get `vcs` errors.

_Originally posted by @brandonwillard in https://github.com/aesara-devs/aesara/discussions/1456#discussioncomment-5133445_

**Comment by PaulScemama** [comment]
And as well, we should change the upstream url to `https://` instead of `git://`?

**Comment by brandonwillard** [comment]
> And as well, we should change the upstream url to `https://` instead of `git://`?

That sounds more like a local configuration detail, but we could add it as a side note.

**Comment by PaulScemama** [comment]
@brandonwillard documenting my experiences:

Following the developer guide [here](https://aesara.readthedocs.io/en/latest/dev_start_guide.html), the steps I needed to run *to yield no errors* were as follows:

* `git clone [forked repository]`
* `git remote add upstream https://github.com/aesara-devs/aesara.git` 
   * `https` was the only one that worked for me. 
   * `ssh` did not work (permission issues) and `git` will result in an error when fetching tags. 
* `git fetch -t --all`
  * just doing `git fetch -t` didn't seem to work. 
* `conda env create -n aesara-dev -f environment-arm.yml`
  * `-arm` because I'm using an M1 Mac
* `conda activate aesara-dev`
* `pip install -r requirements.txt`
* `pre-commit install`

Trying to think of general way to phrase the docs: something like
* `git clone [forked repository]`
* `git remote add upstream git://github.com/aesara-devs/aesara.git`
* `git fetch -t --all`
  * If an error occurs here about `errno=Operation timed out`, then you may need to run `git remote set-url upstream https://github.com/aesara-devs/aesara.git`
  * run `git fetch -t --all` again.
* `conda env create -n aesara-dev -f environment.yml`
  * `environment-arm.yml` if using an ARM processor
* etc.

Apologies for the verbosity. Let me know what you think.

**Comment by dgerlanc** [comment]
This looks great! Appreciate the detailed instructions.

**Comment by brandonwillard** [comment]
> Apologies for the verbosity. Let me know what you think.

I agree with @dgerlanc; your walkthrough looks great.  Feel free to open a PR for this at any time.

**Comment by PaulScemama** [comment]
@dgerlanc @brandonwillard thank you! I will most likely open a PR this coming Monday.
